system_metrics_collector/test/system_metrics_collector/test_composition.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "class_loader/class_loader.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/node_factory.hpp"
+
+#include "test_constants.hpp"
+
+namespace
+{
+constexpr const char kSystemMetricsCollectorLibName[] = "libsystem_metrics_collector.so";
+constexpr const std::array<const char *, 2> kExpectedClassNames = {
+  "system_metrics_collector::LinuxProcessCpuMeasurementNode",
+  "system_metrics_collector::LinuxProcessMemoryMeasurementNode"
+};
+}
+
+bool IsExpectedClassName(const std::string & class_name)
+{
+  for (auto expected_class_name : kExpectedClassNames) {
```

Comment:
Why not use `std::find` instead of using a loop?
```c++
auto result = std::find(kExpectedClassNames.begin(), kExpectedClassNames.end(), class_name);
return result != kExpectedClassNames.end();
```

Reply:
The elements in `kExpectedClassNames` will actually be substrings of `class_name`.

Reply:
But sure, I can replace the `for` loop with a `std::find_if()`.